### PR TITLE
Split Retrofit stuff into own module, new module for Retrofit2

### PR DESCRIPTION
**Pull request by robfletcher**
This splits the Retrofit and OkHttp configuration from `kork-web` and into their own modules; `kork-okhttp3` and `kork-retrofit-legacy`. 
There's also now a new `kork-retrofit` module for Retrofit 2.

### CODE REVIEW

File: kork-retrofit-legacy/src/main/java/com/netflix/spinnaker/config/OkHttpClientConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import org.springframework.stereotype.Component;
+import static com.squareup.okhttp.ConnectionSpec.MODERN_TLS;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * @deprecated replaced by {@link OkHttp3ClientConfiguration}
+ */
+@Component
+@Deprecated // see OkHttp3ClientConfiguration
+public class OkHttpClientConfiguration {
```
Comment (robfletcher) [inline kork-retrofit-legacy/src/main/java/com/netflix/spinnaker/config/OkHttpClientConfiguration.java]: Wondering if we can just drop this old OkHttp stuff. Are any apps still using it?

Comment (emjburns) [reply kork-retrofit-legacy/src/main/java/com/netflix/spinnaker/config/OkHttpClientConfiguration.java]: I think definitely apps are still using it. I keep running into it all over the place and I'm never sure which to use.

**Comment by emjburns** [comment]
Could you add a readme with some brief words on when to use which classes here (or an example of a class doing this well)? And maybe some words on the deprecation strategy? Because I'm bad at knowing what to use and I want to carry this good work on to other services.

File: kork-retrofit-legacy/src/main/java/com/netflix/spinnaker/config/OkHttpClientConfiguration.java (hunk 1 above)
Comment (ajordens) [reply kork-retrofit-legacy/src/main/java/com/netflix/spinnaker/config/OkHttpClientConfiguration.java]: Happy to be proven wrong but pretty sure most services are still using it.

File: kork-retrofit/src/main/java/com/netflix/spinnaker/config/Retrofit2ConfigurationProperties.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.netflix.spinnaker.config
+
+import okhttp3.logging.HttpLoggingInterceptor
+import org.springframework.boot.context.properties.ConfigurationProperties
+
+@ConfigurationProperties(prefix = "retrofit2")
```
Comment (ajordens) [inline kork-retrofit/src/main/java/com/netflix/spinnaker/config/Retrofit2ConfigurationProperties.kt]: Do we really need the `retrofit` vs `retrofit2` prefixes?

File: kork-okhttp3/kork-okhttp3.gradle [hunk 3]
```diff
@@ -0,0 +1,9 @@
+apply plugin: "java-library"
+
+dependencies {
+  api project(":kork-security")
+  api "com.squareup.okhttp3:okhttp:${spinnaker.version("okHttp3")}"
+  api "org.springframework:spring-context:${spinnaker.version("spring")}"
+  api "org.springframework.boot:spring-boot-autoconfigure:${spinnaker.version("springBoot")}"
+  api "com.netflix.spectator:spectator-api:${spinnaker.version("spectator")}"
```
Comment (ajordens) [inline kork-okhttp3/kork-okhttp3.gradle]: api == the old compile scope?

File: kork-retrofit/kork-retrofit.gradle [hunk 4]
```diff
@@ -0,0 +1,10 @@
+apply plugin: "java-library"
+apply from: "$rootDir/gradle/kotlin.gradle"
+
+dependencies {
+  api project(":kork-security")
+  api project(":kork-okhttp3")
+  api "com.squareup.retrofit2:retrofit:${spinnaker.version("retrofit2")}"
+
+  implementation "com.squareup.okhttp3:logging-interceptor:${spinnaker.version('okHttp3')}"
```
Comment (ajordens) [inline kork-retrofit/kork-retrofit.gradle]: implementation == the old provided scope?

File: kork-okhttp3/kork-okhttp3.gradle (hunk 3 above)
Comment (robfletcher) [reply kork-okhttp3/kork-okhttp3.gradle]: Kind of. `api` is compile + transitively exported. `implementation` is compile + runtime only for consumers, `compileOnly` is not transitively exported at all.

File: kork-retrofit/kork-retrofit.gradle (hunk 4 above)
Comment (robfletcher) [reply kork-retrofit/kork-retrofit.gradle]: See above

File: kork-retrofit/src/main/java/com/netflix/spinnaker/config/Retrofit2ConfigurationProperties.kt (hunk 2 above)
Comment (robfletcher) [reply kork-retrofit/src/main/java/com/netflix/spinnaker/config/Retrofit2ConfigurationProperties.kt]: Unfortunately the log level enum is incompatible between the two, so yeah.

Comment (cfieber) [reply kork-retrofit/src/main/java/com/netflix/spinnaker/config/Retrofit2ConfigurationProperties.kt]: we could take the log level name as a string and do something to produce both enum values?

File: kork-retrofit-legacy/src/main/java/com/netflix/spinnaker/config/OkHttpClientConfiguration.java (hunk 1 above)
Comment (robzienert) [reply kork-retrofit-legacy/src/main/java/com/netflix/spinnaker/config/OkHttpClientConfiguration.java]: All apps are using it still.

File: kork-retrofit/src/main/java/com/netflix/spinnaker/config/Retrofit2ConfigurationProperties.kt (hunk 2 above)
Comment (robzienert) [reply kork-retrofit/src/main/java/com/netflix/spinnaker/config/Retrofit2ConfigurationProperties.kt]: Yes, we can attach a converter to this to generate the value. https://www.baeldung.com/spring-type-conversions

**Comment by robfletcher** [comment]
I rebased this. @robzienert @cfieber can you just eyeball the two metrics interceptors?

**Pull request closed without merging**